Build failure: zssh

## Steps To Reproduce

Steps to reproduce the behavior:

1. build zssh

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/w4yy210dg8kdmb5g7615v3hkarqvp3lw-zssh-1.5c.tgz
source root is zssh-1.5c
setting SOURCE_DATE_EPOCH to timestamp 1064377643 of file zssh-1.5c/fake_readline/Makefile
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/j7i4bbxap22mlh88q8y9jcm0s1mp9p73-fix_use_ptmx_on_arch.patch
patching file configure
Hunk #1 succeeded at 1892 (offset -1 lines).
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./lrzsz-0.12.20/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./lrzsz-0.12.20/config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --prefix=/nix/store/fm3bkl7qjimdfs2wk1jbssqr5j8p4g5j-zssh-1.5c
creating cache ./config.cache
checking whether to use libreadline for cmd-line editing... yes
checking for a BSD compatible install... /nix/store/4s9rah4cwaxflicsk5cndnknqlk9n4p3-coreutils-9.5/bin/install -c
checking for gmake... make
checking for gtar... tar
checking for gar... ar
checking for granlib... ranlib
checking for gcc... gcc
checking whether the C compiler (gcc  ) works... no
configure: error: installation or configuration problem: C compiler cannot create executables.
```

</details>

## Additional context

Was one of two build failures I had when trying to update. The other was `mpack`, which matches [this issue](https://github.com/NixOS/nixpkgs/issues/369083).

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.57, NixOS, 25.05 (Warbler), 25.05.20241213.3566ab7`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.24.10`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/22r7q7s9552gn1vpjigkbhfgcvhsrz68-source`
```

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/zs/zssh/package.nix
+++ b/pkgs/by-name/zs/zssh/package.nix
@@ -3,6 +3,7 @@
   stdenv,
   fetchurl,
   readline,
+  autoreconfHook,
 }:
 
 let
@@ -17,6 +18,7 @@ stdenv.mkDerivation rec {
     sha256 = "06z73iq59lz8ibjrgs7d3xl39vh9yld1988yx8khssch4pw41s52";
   };
 
+  nativeBuildInputs = [ autoreconfHook ];
   buildInputs = [ readline ];
 
   patches = [
@@ -30,9 +32,13 @@ stdenv.mkDerivation rec {
 
   patchFlags = [ "-p0" ];
 
+  postPatch = ''
+    sed -i 1i'#include <pty.h>' openpty.c
+  '';
+
   # The makefile does not create the directories
   postBuild = ''
-    install -dm755 "$out"/{bin,man/man1}
+    install -dm755 "$out"/{bin,share/man/man1}
   '';
 
   meta = {

